drivers/serial/uart_hpmicro.c

```diff
@@ -0,0 +1,322 @@
+/*
+ * Copyright (c) 2022 hpmicro
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ */
+
+#define DT_DRV_COMPAT hpmicro_hpm_uart
+
+#include <errno.h>
+#include <drivers/pinctrl.h>
+#include <drivers/uart.h>
+#include <soc.h>
+#include "hpm_common.h"
+#include "hpm_uart_drv.h"
+#include "hpm_clock_drv.h"
+
+struct uart_hpm_cfg {
+	UART_Type *base;
+	uint32_t clock_name;
+	uint32_t clock_src;
+	uint32_t parity;
+#ifdef CONFIG_UART_INTERRUPT_DRIVEN
+	uart_irq_config_func_t irq_config_func;
+#endif /* CONFIG_UART_INTERRUPT_DRIVEN */
+#ifdef CONFIG_PINCTRL
+	const struct pinctrl_dev_config *pincfg;
+#endif /* CONFIG_PINCTRL */
```

Comment:
it doesn't seem your platform has an alternative to pinctrl, so such ifdef should be removed.